PyBASC/pipeline.py

```diff
@@ -418,3 +423,511 @@ def create_basc(proc_mem, name='basc'):
+        outputspec.cluster_voxel_scores: ndarray
+            Group stability map using gsm and gscluster to calculate average
+            within-cluster stability
+        outputspec.gsclusters_img : string (nifti file)
+            3-D volume of brain regions partitioned with gsclusters
+        outputspec.cluster_voxel_scores_img : string (nifti file)
+            3-D volume of brain regions associated with gs_map
+
+    BASC Procedure:
+
+    1. Generate individual stability matrices based on multiple clusterings of
+       each bootstrap sample for a single subject
+    2. Use stratified bootstrap to sample new datasets of subjects
+    3. Calculate average stability matrix of each new dataset using individual
+       stability matrices generated at step 1
+    4. Cluster each average stabiilty matrix
+    5. Average to create a group stability matrix
+    6. Cluster the group stability matrix
+    7. Calculate average within-cluster stability based on clustering of step 6
+
+    Workflow Graph:
+
+    .. image:: ../images/basc.dot.png
+        :width: 500
+
+    Detailed Workflow Graph:
+
+    .. image:: ../images/basc_detailed.dot.png
+        :width: 500
+
+    References
+    ----------
+    .. [1] P. Bellec, P. Rosa-Neto, O. C. Lyttelton, H. Benali, and A. C. Evans,
+       "Multi-level bootstrap analysis of stable clusters in resting-state fMRI"
+       NeuroImage, vol. 51, no. 3, pp. 1126-39, Jul. 2010.
+
+    Examples
+    --------
+    >>> from CPAC import basc
+
+    """
+
+    mem_per_proc = float(proc_mem[1]) / float(proc_mem[0])
+
+    basc_wf = pe.Workflow(name=name)
+
+    inputspec = pe.Node(util.IdentityInterface(fields=[
+        'subjects_files',
+        'roi_mask_file',
+        'cross_cluster',
+        'cxc_roi_mask_file',
+        'group_dim_reduce',
+        'random_state_tuple',
+    ]), name='inputspec')
+
+    inputspec_compression_dim = pe.Node(
+        util.IdentityInterface(fields=[
+            'compression_dim',
+        ]),
+        name='inputspec_compression_dim'
+    )
+
+    inputspec_boostraps = pe.Node(
+        util.IdentityInterface(fields=[
+            'dataset_bootstraps',
+            'timeseries_bootstraps',
+        ]),
+        name='inputspec_boostraps'
+    )
+    inputspec_boostraps.synchronize = True
+
+    inputspec_similarity_metric = pe.Node(
+        util.IdentityInterface(fields=[
+            'similarity_metric',
+        ]),
+        name='inputspec_similarity_metric'
+    )
+
+    inputspec_cluster_method = pe.Node(
+        util.IdentityInterface(fields=[
+            'cluster_method',
+        ]),
+        name='inputspec_cluster_method'
+    )
+
+    inputspec_blocklength = pe.Node(
+        util.IdentityInterface(fields=[
+            'blocklength',
+        ]),
+        name='inputspec_blocklength'
+    )
+
+    inputspec_n_clusters = pe.Node(
+        util.IdentityInterface(fields=[
+            'n_clusters',
+        ]),
```

Comment:
Create nipype graph (detailed flat) visualization of pipeline in both regular and optimized cases